index.js

```diff
@@ -812,66 +814,79 @@ class Client {
    * @param {Object} opts options that passed directly to request
    * @param {string} opts.method The HTTP method to be called (default 'GET')
    * @param {Object} opts.qs An object of query parameters to be passed to the backend
-   * @param {string} opts.body The body of the request (for POST requests only)
+   * @param {string} opts.body The body of the request (for post requests only)
    * @returns {Promise<Response>} A promise of the response
    */
+
   request(path, opts) {
     const uri = this.baseUrl + path;
-    const params = Object.assign(
+    let configuration = Object.assign(
       {
-        method: 'GET',
-        uri: uri,
+        url: uri,
+        method: "get",
         json: true,
         gzip: true,
+        timeout: 4000,
       },
       opts
     );
-    return rp(params).catch((e) => {
-      console.error(`Error in API ${uri}: Status ${e.statusCode}`);
-      throw e;
-    });
+
+    if (configuration.qs) {
+      configuration = Object.assign(configuration, {
+        params: configuration.qs,
+      });
+      delete configuration.qs;
+    }
+
+    return axios(configuration)
+      .then((res) => {
+      return res})
```

Comment:
```suggestion
          return res.data
      })
```

[From the axios documentation](https://www.npmjs.com/package/axios#response-schema), we know that we want the data property here. We are not making use of other properties, so making this one change here would eliminate the need to make many other changes.

Reply:
Keeping it generic to match the post request response and destructuring data for get request.